### PR TITLE
Added Azure Functions

### PR DESCRIPTION
**Pull request by zippy1981**
### What does this PR do?
Add Azure functions to the list of instrumentations.

### Motivation
The list is incomplete, but the list is also divided between .NET framework and .NET core so I am opening a separate issue to talk about how we can simplify that.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by buraizu** [comment]
Thanks, created DOCS-3744 to track this update

File: content/en/tracing/setup_overview/compatibility_requirements/dotnet-core.md [hunk 1]
```diff
@@ -54,31 +54,32 @@ The .NET Tracer supports automatic instrumentation on the following architecture
...
-| WebClient / WebRequest          | `System.Net.Requests` 4.0+                                                                | `WebRequest`         |
+| Framework or library            | NuGet package                                                                                        | Integration Name     |
+| ------------------------------- | ---------------------------------------------------------------------------------------------------- | -------------------- |
+| ADO.NET                         | All AdoNet integrations                                                                              | `AdoNet`             |
+| Aerospike                       | `Aerospike.Client` 4.0.0+                                                                            | `Aerospike`          |
+| ASP.NET Core                    | `Microsoft.AspNetCore`</br>`Microsoft.AspNetCore.App`</br>2.0+ and 3.0+                              | `AspNetCore`         |
+| Azure Functions                 | `Microsoft.Azure.Webjobs` 3.0+                                                                       | `AzureFunctions`     |
```
Comment (lucaspimentel) [inline content/en/tracing/setup_overview/compatibility_requirements/dotnet-core.md:62]: @pierotibou @andrewlock, what do you think of including Azure Functions here? It's a little different than other integrations, so I'm not sure.

Comment (pierotibou) [reply content/en/tracing/setup_overview/compatibility_requirements/dotnet-core.md:62]: LGTM, we do instrument this assembly, it generates a span. Makes sense to be here

**Review by pierotibou — APPROVED** [review]

File: content/en/tracing/setup_overview/compatibility_requirements/dotnet-core.md (hunk 1 above)
Comment (andrewlock) [reply content/en/tracing/setup_overview/compatibility_requirements/dotnet-core.md:62]: I wonder if it's worth pointing out _which_ AzureFunctions libraries we instrument though? e.g. which versions/inproc/outofproc etc (being vague as I can't remember 😂 )

**Review by kayayarai — APPROVED** [review]

**Pull request merged**